src/verify.rs

```diff
@@ -89,15 +89,35 @@ use self::PreparedStep::*;
 ///
 /// This type would be Copy except for the fact that the bitset can require
 /// overflow storage :(.
-struct StackSlot {
+#[derive(Clone)]
+pub struct StackSlot {
     vars: Bitset,
     code: Atom,
     expr: Range<usize>,
 }
 
+/// A constructor trait for plugging in to the verifier, to collect extra data during the
+/// verification pass
+pub trait ProofBuilder {
```

Comment:
The 8-11% runtime regression for one-shot verification is a bit much to take.

I've identified two plausible causes below but I'll need to spend some time to confirm that.

This is not at all how I intended `verify` to be used, but I can't articulate a specific complaint, so.

```
$$ <<1>> = perf stat ./smetamath.noverifychanges --split --jobs 4 --verify ~/dlcode/set.mm/set.mm
$$ <<2>> = perf stat ./smetamath.export-mmp --split --jobs 4 --verify ~/dlcode/set.mm/set.mm
$$ <<3>> = perf stat ./smetamath.master --split --jobs 4 --verify ~/dlcode/set.mm/set.mm
...
 <<1>> seconds time elapsed
-> << 1>> (# 1)  count=100 mean=4.7729e-01 mlow3=4.6572e-01 mhi3=4.8886e-01 sd=3.8565e-02  tcount=81 tmean=4.6924e-01 tmlow3=4.5767e-01 tmhi3=4.8081e-01 tsd=1.436979e-02
-> << 1>> (# 2)  count=100 mean=5.2754e-01 mlow3=5.1096e-01 mhi3=5.4412e-01 sd=5.5272e-02  tcount=81 tmean=5.1448e-01 tmlow3=4.9790e-01 tmhi3=5.3106e-01 tsd=2.187830e-02
-> << 1>> (# 3)  count=100 mean=4.7155e-01 mlow3=4.6283e-01 mhi3=4.8027e-01 sd=2.9057e-02  tcount=81 tmean=4.6498e-01 tmlow3=4.5626e-01 tmhi3=4.7370e-01 tsd=1.190247e-02
```

(I wrote some tools last week to do randomized differential benchmarking.  Not super pretty but it works and gives me more confidence than plain `perf stat --repeat`; mlow3/mhi3 is a 3-sigma confidence interval.)